packages/openapi-v3/src/controller-spec.ts

```diff
@@ -275,3 +276,38 @@ export function getControllerSpec(constructor: Function): ControllerSpec {
   }
   return spec;
 }
+
+function validatePath(path: string = '/') {
+  let tokens = pathToRegExp.parse(path);
+  if (tokens.some(t => typeof t === 'object')) {
+    throw new Error(
+      `Invalid path template: '${path}'. Please use {param} instead of ':param'`,
+    );
+  }
+
+  path = path.replace(/{([^}]*)}(\/|$)/g, ':$1$2');
+  tokens = pathToRegExp.parse(path);
+  for (const token of tokens) {
+    if (typeof token === 'string') continue;
+    if (typeof token.name === 'number') {
+      // Such as /(.*)
+      throw new Error(`Unnamed parameter is not allowed in path '${path}'`);
+    }
+    if (token.optional || token.repeat || token.pattern !== '[^\\/]+?') {
+      // Such as /:foo*, /:foo+, /:foo?, or /:foo(\\d+)
+      throw new Error(`Parameter modifier is not allowed in path '${path}'`);
+    }
+  }
+  return path;
+}
+
+/**
+ * Validate and normalize the path to OpenAPI path template. No parameter
+ * modifier, custom pattern, or unnamed parameter is allowed.
+ *
+ * @param path Express style path (https://github.com/pillarjs/path-to-regexp)
+ */
+export function toOpenApiPath(path: string = '/') {
+  validatePath(path);
+  return path.replace(/\:([A-Za-z0-9_]+)/g, '{$1}');
```

Comment:
Is this `replace` call still needed, considering that `validatePath` rejects values containing express-style parameters? Maybe I am missing something?